plugins/flatpak/gs-flatpak.c

```diff
@@ -3396,6 +3411,182 @@ gs_flatpak_create_app_from_repo_dir (GsFlatpak *self,
 	return app;
 }
 
+gboolean
+gs_flatpak_app_get_copyable (GsFlatpak *self,
+			     GsApp *app,
+			     gboolean *copyable,
+			     GCancellable *cancellable,
+			     GError **error)
+{
+	if (copyable) {
+		*copyable = (gs_flatpak_get_collection_id (self, app) != NULL);
+	}
+
+	return TRUE;
+}
+
+typedef struct {
+	GsFlatpak *self;
+	GsApp *app;
+	GCancellable *cancellable;
+	gulong cancelled_id;
+	gboolean finished;
+	GError *error;
+} GsFlatpakCopyProcessHelper;
+
+static void
+gs_flatpak_copy_process_helper_free (GsFlatpakCopyProcessHelper *helper)
+{
+	g_object_unref (helper->app);
+	g_object_unref (helper->cancellable);
```

Comment:
Nitpick: Maybe it's better/safer to clear it these objects instead?

Reply:
Sure. Fixed locally.